src/components/SiteInfoInputForm.js

```diff
@@ -0,0 +1,786 @@
+import React, {Component} from 'react';
+import {Alert, Picker, Platform, ScrollView, View, Image} from 'react-native';
+import {connect} from 'react-redux';
+import {Icon, Text, CheckBox, Input} from 'react-native-elements';
+
+import _ from 'lodash';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import {LargeButton, SmallButton} from './common';
+
+import {
+    updateLatitudeText,
+    updateLongitudeText,
+    updateSiteTitleText,
+    updateSiteDescriptionText,
+    updateSiteDirectionsText,
+    updateSiteNearestTownText,
+    updateAccessibilityOption,
+    updatePriceOption,
+    promptForLocationServicesPermission,
+    promptForGalleryPermission,
+    launchPhotoGallery,
+    launchCamera,
+    promptForCameraPermission,
+    getCurrentUserLocation,
+    checkIfSiteIsReadyForUpload,
+    attemptToUploadSite,
+    siteDetailCheckboxWasClicked,
+    updateAlternateSitesText,
+    updateCellProviderOption,
+    updateCellStrengthOption,
+    updateCountyOption,
+    updateForestOption,
+    updateMVUMOption,
+    newSiteToEditAvailable
+} from '../actions';
+
+import {campsite, submit_form, common, more_screen, counties, mvum_names, forest_names} from '../locale.en';
+
+const {
+    campsite_form: {
+        reset,
+        site_image,
+        latitude, longitude,
+        longitude_placeholder, latitude_placeholder,
+        add_site, add_a_campsite, add_site_title, site_info,
+        description, description_placeholder,
+        directions, directions_placeholder,
+        optional,
+        alternate_sites, alternate_sites_placeholder,
+        nearest_town, nearest_town_placeholder,
+        here_now,
+        accessibility, facilities, price, features, cell_service, county, forest, mvum,
+        accessibility_options,
+        facilities_options,
+        features_options,
+        price_options,
+        cell_provider_options,
+        cell_strength_options
+    }
+} = campsite;
+
+const {submit} = submit_form;
+
+const {title, location} = common;
+
+import {map, navKeys, permissionResponses, site_form_type} from '../constants';
+
+const {GRANTED, DENIED, UNDETERMINED} = permissionResponses;
+
+import {navyBlueButton, grey, darkBlue, inputLabel, approveGreen, rejectRed} from '../styles';
+
+class SiteInfoInputForm extends Component {
+
+    componentWillMount() {
+        const {siteFormType} = this.props;
+
+        this.props.checkIfSiteIsReadyForUpload({siteFormType});
+
+        if(siteFormType === site_form_type.EDIT){
+            const {siteToEdit} = this.props;
+
+            this.props.newSiteToEditAvailable({siteToEdit});
+        }
```

Comment:
If the form we're loading is an edit, we should pre-populate all of the fields with the `siteToEdit` that was passed as a prop.